test/python/synthesis/test_siswap_synthesis.py

```diff
@@ -0,0 +1,137 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Test synthesis of two-qubit unitaries into SiSwap gates."""
+
+import unittest
+from test import combine
+from ddt import ddt
+import numpy as np
+
+from qiskit.synthesis.su4 import SiSwapDecomposer
+from qiskit.quantum_info import random_unitary, Operator, average_gate_fidelity
+from qiskit.test import QiskitTestCase
+from qiskit.circuit.library import (
+    SwapGate,
+    iSwapGate,
+    CXGate,
+    IGate,
+    XGate,
+    RXXGate,
+    RYYGate,
+    RZZGate,
+)
+
+_EPS = 1e-12
+
+
+@ddt
+class TestSiSwapSynth(QiskitTestCase):
+    """Test synthesis of SU(4)s over SiSwap basis."""
+
+    @combine(seed=range(50))
+    def test_siswap_random(self, seed):
+        """Test synthesis of 50 random SU(4)s."""
+        u = random_unitary(4, seed=seed)
+        decomposer = SiSwapDecomposer(euler_basis=["rz", "ry"])
+        circuit = decomposer(u)
+        self.assertLessEqual(circuit.count_ops().get("siswap", 0), 3)
+        self.assertEqual(Operator(circuit), Operator(u))
```

Comment:
In all these in this file, should we also be testing that the only gates in the output are those in the Euler basis as well?